care/users/api/viewsets/users.py

```diff
@@ -253,6 +253,12 @@ def clear_home_facility(self, request, *args, **kwargs):
         if not self.has_user_type_permission_elevation(requesting_user, user):
             raise ValidationError({"home_facility": "Cannot Access Higher Level User"})
 
+        # ensure that district admin only able to delete in the same district
+        if user.district_id != requesting_user.district_id:
```

Comment:
A state admin need not have the same district as the requesting user right? Superadmin would not need any checks  as well.

Reply:
@vigneshhari done, now this will only check for users equal to or below District Admin's position